documentation/release_notes.markdown

```diff
@@ -61,179 +94,253 @@ This release retires all api versions before v4 and promotes v4 to be
 
-* Add Apache-style access logging to PuppetDB packages, and enable it by default  ([PDB-477](https://tickets.puppetlabs.com/browse/PDB-477))
+#### API changes
 
-* Add a `--url-prefix` option to the import and export commands in the PuppetDB CLI  ([PDB-1068](https://tickets.puppetlabs.com/browse/PDB-1068))
+* The query parameter for `/pdb/query/v4/events` is now optional ([PDB-1132](https://tickets.puppetlabs.com/browse/PDB-1132))
 
-    This allows import and export to work against a PuppetDB configured with a url prefix.
+* Made `/pdb/query/v4/catalogs` queryable ([PDB-1028](https://tickets.puppetlabs.com/browse/PDB-1028))
 
-* Add the `[puppetdb] disable-update-checking` configuration key ([PDB-158](https://tickets.puppetlabs.com/browse/PDB-158))
+* Added `producer\_timestamp` field to factsets and catalogs ([PDB-489](https://tickets.puppetlabs.com/browse/PDB-489))
 
-    The value defaults to false; see the [configuration](./configure.html) documentation for more details.
+    On factsets the field is queryable and orderable.
 
-* Support fallback PuppetDB connections ([PDB-954](https://tickets.puppetlabs.com/browse/PDB-954))
+* Added `producer\_timestamp` field to reports ([PDB-1487](https://tickets.puppetlabs.com/browse/PDB-1487))
 
-    When configuring your connection to PuppetDB, you can now specify a fallback.
-    Instead of the old `server` and `port` config, you can now use a
-    `server_urls` config that contains the full path to one or more PuppetDB
-    URLs. The old server/port format is still supported.
+* Added `resource\_events` key to the `store report` command ([PDB-1072](https://tickets.puppetlabs.com/browse/PDB-1072))
 
-    This commit also supports a new `server_url_timeout` config. If the
-    PuppetDB instance has not responded in the specified number of
-    seconds (defaults to 30) it will fail and roll to the next PuppetDB
-    url (if one is provided).
+    API responses can now be easily used for resubmission as a
+    report. This will be helpful in the HA context, where it will reduce
+    the number of requests needed to synchronize reports.
 
-* Add factset hash to `replace-facts!` ([PDB-898](https://tickets.puppetlabs.com/browse/PDB-898))
+* The command `deactivate node` now requires a `producer_timestamp` field ([PDB-1310](https://tickets.puppetlabs.com/browse/PDB-1310))
 
-    This commit associates a hash of each factset with the metadata of the factset
-    to allow us to easily tell if fact values have changed.
+#### Performance
+
+* Switch from `text` based storage of UUID and hashes, to the PostgreSQL `uuid` and `bytea` types respectively ([PDB-1416](https://tickets.puppetlabs.com/browse/PDB-1416))
+
+* The `reports` database table was using the hash string as its
+  primary key, we know have switched to using a smaller bigint primary
+  key for that table, which should result in faster joins in most
```

Comment:
is "should" really accurate? Is there a case where these things wouldn't be true?